src/nvim/eval/decode.c

```diff
@@ -973,7 +973,13 @@ int msgpack_to_vim(const msgpack_object mobj, typval_T *const rettv)
       }
       break;
     }
-    case MSGPACK_OBJECT_FLOAT: {
+#ifdef NVIM_MSGPACK_HAS_FLOAT32
```

Comment:
I generally prefer "feature" checks instead of version checks.  It's not as relevant here, but they're more robust in terms of changes being backported.